src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs

```diff
@@ -176,6 +176,18 @@ public bool AreEqual(object? x, object? y, ref Tolerance tolerance)
         {
             EqualMethodResult result = AreEqual(x, y, ref tolerance, new ComparisonState(true));
 
+            return ProcessEqualResult(x, y, result);
+        }
+
+        internal bool AreEqual<T>(T? x, T? y, ref Tolerance tolerance)
```

Comment:
Something like this, if left in the final design, could also lend itself well to #3985